### ISSUE TITLE
Project Status

### ISSUE THREAD

**Issue by DrSensor** [kaitai-io/kaitai_struct_rust_runtime#4]
Hi, whats the status of this project?
I found something similar which use compile time macro to convert kaitai yaml into rust code: https://github.com/manuels/taikai
Is there any conformance test suite for kaitai?

**Comment by bspeice** [comment]
The project has been abandoned, I'll set it to archive status. Kaitai tests are automatically generated, so it's possible to re-use that infrastructure.

**Comment by bspeice** [comment]
EDIT: I had an earlier comment that was incorrect, given I erroneously assumed it was on a repository I maintain with the same name. Please ignore.

**Comment by TwoClocks** [comment]
I think this is abandoned. The output of the compiler doesn't compile under 2021 addition, etc.

**Comment by gbonnefille** [comment]
@generalmimon can you comment on this issue as you actively maintain the project?

**Comment by gant-topcon** [comment]
I'd also like to know the maintenance status of the project as this will determine how much I can rely on Kaitai Rust support at work. Here are some problems I spotted:

* The official [Kaitai Stream API docs](https://doc.kaitai.io/stream_api.html) do not mention the rust API or crate. However, on the [kaitai.io](https://kaitai.io/) front page, Rust is listed as a supported language without disclaimers.
* The Rust output of a `.ksy` file using the `kaitai-struct-compiler 0.11` will create a Rust source file containing `extern crate kaitai`, which is highly misleading, since **this is not the kaitai crate on crates.io**. Only using the `kaitai_struct_rust_runtime` crate and aliasing it in the `Cargo.toml` file [as demonstrated in this comment](https://github.com/kaitai-io/kaitai_struct_rust_runtime/issues/9) works.

If this is just a documentation issue, I'm willing to help contribute to the docs. Please reach out if this is of interest. 

If however, this is a support issue, I'd appreciate if the front-page added some disclaimers to the Rust part.

**Comment by generalmimon** [comment]
@gant-topcon:

> * The official [Kaitai Stream API docs](https://doc.kaitai.io/stream_api.html) do not mention the rust API or crate.

I agree that the discoverability of the runtime libraries in general could be better (see https://github.com/kaitai-io/kaitai-io.github.io/issues/20). It seems that some people trying out Kaitai Struct for the first time are not aware of the fact that they need them at all or where to find them. I would think that some instructions that I added at individual format pages at https://formats.kaitai.io/ in https://github.com/kaitai-io/kaitai_struct/issues/571 probably helped at least someone in that regard, but that can only work if they land on a page like https://formats.kaitai.io/elf/python.html and read it, and these instructions are also not implemented in all target languages. I'm not sure where to put this information specifically, though. "KS runtime library" is already briefly mentioned in https://kaitai.io/#what-is-it, but that's apparently not enough...

Also, after being involved in this project for many years, I don't really see the purpose of the https://doc.kaitai.io/stream_api.html page. It looks like it only contains information useful for Kaitai developers, so I guess that's supposed to help me, but I've never had a reason to look there. I always look at the source code of individual runtime libraries directly, since that's the only authoritative source. Perhaps we should just delete https://doc.kaitai.io/stream_api.html to avoid confusion.

> * The Rust output of a `.ksy` file using the `kaitai-struct-compiler 0.11` will create a Rust source file containing `extern crate kaitai`, which is highly misleading, since **this is not the kaitai crate on crates.io**. Only using the `kaitai_struct_rust_runtime` crate and aliasing it in the `Cargo.toml` file [as demonstrated in this comment](https://github.com/kaitai-io/kaitai_struct_rust_runtime/issues/9) works.

Yes, this is a legitimate problem. I realized this when I wanted to publish the 0.11 Rust runtime library to crates.io, only to find out that it wasn't possible because the name "kaitai" was already taken. And since I had already published KSC 0.11, I couldn't change the hardcoded `extern crate kaitai` declarations in the generated code, and I still can't do that.

So for the time being, all we can do is state in the README that https://github.com/kaitai-io/kaitai_struct_rust_runtime/issues/9 is necessary.

> If this is just a documentation issue, I'm willing to help contribute to the docs.

Please do.

**Comment by gant-topcon** [comment]
Hey @generalmimon, thanks for getting back to me and good to see the project is still alive. What's the best way to start contributing? Just open PRs or should I get in touch with someone?

**Comment by generalmimon** [comment]
@gant-topcon:

> What's the best way to start contributing? Just open PRs or should I get in touch with someone?

Opening PRs is fine. For larger/unclear features, it's probably better to open an issue in advance to reduce the risk of rejection if it doesn't meet our expectations (of course, this can always happen in any project, although we don't do it lightly).
